lib/clouds/shared_functions.py

```diff
@@ -813,13 +822,342 @@ def populate_interface(self, obj_attr_list) :
+
+                            if self.get_description() == "DigitalOcean Cloud" :
+                                connection.delete_key_pair(_registered_key_pair_objects[key_name])                                
+                            break
+
+            if not _key_pair_found :
+
+                _msg = "    Creating the ssh key pair \"" + key_name + "\""
+                _msg += " on VMC " + vmc_name + ", using the public key \""
+                _msg += _pub_key_fn + "\"..."
+                
+                if not internal :
+                    cbdebug(_msg, True)
+                else :
+                    cbdebug(_msg)                
+
+                if self.get_description() == "Amazon Elastic Compute Cloud" :
+                    self.ec2conn.import_key_pair(key_name, _key_type + ' ' + _key_contents)
+
+                if self.get_description() == "OpenStack Cloud" :
+                    self.oskconncompute.keypairs.create(key_name, \
+                                                        public_key = _key_type + ' ' + _key_contents)                
+
+                if self.get_description() == "SoftLayer Cloud" :
+                    self.sshman.add_key(_key_type + ' ' + _key_contents, key_name)
+
+                if self.get_description() == "Google Compute Engine" :
+                    _temp_key_metadata[key_name] = key_name + ':' + _key_type + ' ' + _key_contents + " cbtool@orchestrator"
+                    
+                    _key_list_str = ''
+                    for _key in _temp_key_metadata.keys() :
+                        _key_list_str += _temp_key_metadata[_key] + '\n'
+
+                    _key_list_str = _key_list_str[0:-1]
+
+                    if "items" in _metadata["commonInstanceMetadata"] :                    
+                        for _element in _metadata['commonInstanceMetadata']['items'] :
+                            if _element["key"] == "sshKeys" :
+                                _element["value"] = _key_list_str
+                    else :
+                        _metadata['commonInstanceMetadata']['items'].append({"key": "sshKeys", "value" : _key_list_str})
+                                                
+                    self.gceconn.projects().setCommonInstanceMetadata(project=self.instances_project, body=_metadata["commonInstanceMetadata"]).execute(http = self.http_conn[connection])
+
+                if self.get_description() == "DigitalOcean Cloud" :
+                    connection.create_key_pair(key_name, _key_type + ' ' + _key_contents + " cbtool@orchestrator")
+                    
+                _key_pair_found = True
+
+            return _key_pair_found    
+
+    @trace
+    def check_security_group(self,vmc_name, security_group_name) :
+        '''
+        TBD
+        '''
+    
+        _security_group_name = False
+        
+        if security_group_name :
```

Comment:
That is bad form, but in my evaluation is less egregious than the previous two. Would you be willing to let this one slide for now?